src/modules/simulation/gz_bridge/GZBridge.cpp

```diff
@@ -395,7 +400,11 @@ void GZBridge::poseInfoCallback(const ignition::msgs::Pose_V &pose)
 	for (int p = 0; p < pose.pose_size(); p++) {
 		if (pose.pose(p).name() == _model_name) {
 
-			const uint64_t time_us = (pose.header().stamp().sec() * 1000000) + (pose.header().stamp().nsec() / 1000);
+#if defined(ENABLE_LOCKSTEP_SCHEDULER)
+			const uint64_t time_us = (imu.header().stamp().sec() * 1000000) + (imu.header().stamp().nsec() / 1000);
```

Comment:
In this function `time_us` is used for both calculating a dt and setting timestamp_sample in publications.

I think we should continue using the IMU header for the dt calculation regardless.

Reply:
Well, the problem is, that the simulation doesn't work at all if you compile it with the CONFIG_BOARD_NOLOCKSTEP=y flag.

Reply:
True, I make the correction